java/core/src/test/java/co/worklytics/psoxy/rules/RulesBaseTestCase.java

```diff
@@ -153,4 +154,35 @@ protected void assertContains(String content, String... shouldContain) {
             });
     }
 
+    @SneakyThrows
+    protected void assertUrlWithQueryParamsAllowed(String url) {
+        assertTrue(sanitizer.isAllowed(new URL(url + "?param=value")), "single param blocked");
+        assertTrue(sanitizer.isAllowed(new URL(url + "?param=value&param2=value2")), "multiple params blocked");
+    }
+
+
+
+    @SneakyThrows
+    protected void assertUrlWithQueryParamsBlocked(String url) {
+        assertFalse(sanitizer.isAllowed(new URL(url + "?param=value")), "query param allowed");
+        assertFalse(sanitizer.isAllowed(new URL(url + "?param=value&param2=value2")), "multiple query params allowed");
+    }
+
+    @SneakyThrows
+    protected void assertUrlWithSubResourcesAllowed(String url) {
+        assertTrue(sanitizer.isAllowed(new URL(url + "/dsfasdf")), "single param blocked");
+        assertTrue(sanitizer.isAllowed(new URL(url + "/asdfasdf/asdfasdf")), "multiple params blocked");
```

Comment:
Improve readability and nicer for OSS project. Same below.

```suggestion
        assertTrue(sanitizer.isAllowed(new URL(url + "/anypath")), "path blocked");
        assertTrue(sanitizer.isAllowed(new URL(url + "/anypath/anysubpath")), "path with subpath blocked");
```